nni/nas/oneshot/pytorch/strategy.py

```diff
@@ -83,6 +85,80 @@ def export_top_models(self, top_k: int = 1) -> list[Any]:
             warnings.warn('One-shot strategy currently only supports exporting top-1 model.', RuntimeWarning)
         return [self.model.export()]
 
+    @staticmethod
+    def sub_state_dict(model, super_state_dict):
+        """
+        Truncate the state dict of the fixed subnet from that of the search space.
+
+        For example, when you already have a state dict for the base model / search space (which often
+        happens when you have trained a supernet with one-shot strategies), the state dict isn't organized
+        in the same way as when a sub-model is sampled from the search space. This patch will help
+        the modules in the sub-model find the corresponding name and param in the base model.
+        """
+        assert isinstance(model, nn.Module), 'PyTorch is the only supported framework for now.'
```

Comment:
I have another proposal that model is an arch dict here. But I haven't considered pros and cons carefully.

In evaluator:

```python
def evaluate_model_acc(model):
  current_arch_dict = nni.get_current_parameter()['mutation_summary']
  random_oneshot_strategy = strategy.RandomOneShot()
  random_oneshot_strategy.attach_model_space(ModelSpace())
  random_oneshot_strategy.load_state_dict(torch.load('oneshot_strategy.ckpt'))
  current_arch_ckpt = random_oneshot_strategy.subnet_state_dict(current_arch_dict)
  model.load_state_dict(current_arch_dict)
  # evaluate...
```

Your design looks also fine. You can keep it for now.